rke2-rancher-kvm-tf/rancher_provisioner.sh

```diff
@@ -0,0 +1,129 @@
+#!/bin/bash
+
+rancher_version=$1
+install_rke2_version=$2
+
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+NC='\033[0m' # No Color
+
+print_success() {
+  printf "${GREEN}$1${NC}\n"
+}
+
+print_error() {
+  printf "${RED}$1${NC}\n"
+}
+
+
+# Function to wait until a command is successful
+wait_for_success() {
+  local command=$1
+  local max_attempts=$2
+  local delay=$3
+
+  printf "Waiting for command to succeed: $command\n"
+
+  local attempts=0
+  while true; do
+    if eval "$command"; then
+      print_success "Command succeeded: $command\n"
+      break
+    else
+      attempts=$((attempts + 1))
+      if [[ $attempts -ge $max_attempts ]]; then
+        print_error "Exceeded maximum attempts. Command failed: $command\n"
+        exit 1
+      fi
+      sleep "$delay"
+    fi
+  done
+}
+
+# Display a blank line
+log() {
+  local message=$1
+  echo "$(date '+%Y-%m-%d %H:%M:%S') - $message"
+}
+
+# Add Helm repositories for cert-manager 
+log "Adding Helm repositories for cert-manager"
+helm repo add jetstack https://charts.jetstack.io
+helm repo update
+
+log "copy kubeconfig file"
+mkdir -p ~/.kube
+cp -p /etc/rancher/rke2/rke2.yaml ~/.kube/config
+
+
+# Create namespaces for cert-manager and Rancher
+log "Creating namespaces"
+kubectl create namespace cert-manager
+kubectl create namespace cattle-system
+
+# Wait for 1 second
+sleep 1
+
+# Install Cert Manager using Helm
+log "Installing Cert Manager crds"
+kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v1.11.1/cert-manager.crds.yaml
+
+# Add Helm repositories for cert-manager
+log "Adding Helm repositories for cert-manager"
+helm repo add jetstack https://charts.jetstack.io
+helm repo update
+
+log "Installing Cert Manager"
+helm install cert-manager jetstack/cert-manager   --namespace cert-manager   --create-namespace   --version v1.11.1
```

Comment:
will we have to update the cert-manager version here with increase of Rancher or RKE version? regardless, can this be made into a var?